### PR TITLE
Add tests for `hds-icon-registry` service

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

If merged, this PR adds unit tests for the `hds-icon-registry` service.

### :link: External links

Jira ticket: [HDS-6118](https://hashicorp.atlassian.net/browse/HDS-6118)

***

### :eyes: Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

<details>

<summary>:clipboard: PCI review checklist</summary>

- [ ] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.
- [ ] If applicable, I've worked with GRC to document the impact of any changes to security controls.
  Examples of changes to controls include access controls, encryption, logging, etc.
- [ ] If applicable, I've worked with GRC to ensure compliance due to a significant change to the in-scope PCI environment.
  Examples include changes to operating systems, ports, protocols, services, cryptography-related components, PII processing code, etc.

</details>

[HDS-6118]: https://hashicorp.atlassian.net/browse/HDS-6118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: showcase/tests/unit/services/hds-icon-registry-test.ts [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+} from '@hashicorp/design-system-components/services/hds-icon-registry';
+import {
+  HdsIconLibraryValues,
+  HdsIconSizeValues,
+  type HdsIconLoader,
+} from '@hashicorp/design-system-components/components/hds/icon/types';
+
+import type { IconName } from '@hashicorp/flight-icons/svg/index';
+
+const ROOT_ID = 'flight-sprite-empty-container';
```
Comment (didoo) [inline showcase/tests/unit/services/hds-icon-registry-test.ts]: [suggestion] you may consider re-exporting `ROOT_ID` in `hds-icon-registry` and avoid re-defining it here

File: showcase/tests/unit/services/hds-icon-registry-test.ts
Comment (didoo) [inline showcase/tests/unit/services/hds-icon-registry-test.ts:1]: Consider exporting `makeDomSafeId` and `makeSymbolIdFromKey` in the `hds-icon-registry` and add tests for these functions as well

File: showcase/tests/unit/services/hds-icon-registry-test.ts [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+  library,
+  name,
+  size,
+}: HdsIconDefinition): string => {
+  const key =
+    library === HdsIconLibraryValues.Flight
+      ? `${library}-${name}-${size}`
+      : `${library}-${name}`;
+
+  return `hds-icon-${key.replace(/[^a-zA-Z0-9_-]/g, '-')}`;
```
Comment (didoo) [inline showcase/tests/unit/services/hds-icon-registry-test.ts]: if you export `makeDomSafeId` and `makeSymbolIdFromKey` from `hds-icon-registry.ts` then you can use hete here instead of the string concatenation and the regex (which is just a copy, so if one day we change the regex in that function, this test may pass anyway); plus you can test the functions themselves too (see comment above)

File: showcase/tests/unit/services/hds-icon-registry-test.ts [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+    const symbolId = makeSymbolIdFromDefinition(definition);
+    const root = ensureRoot();
+    const existing = document.createElementNS(
+      'http://www.w3.org/2000/svg',
+      'symbol',
+    );
+
+    existing.id = symbolId;
+
+    root.appendChild(existing);
```
Comment (didoo) [inline showcase/tests/unit/services/hds-icon-registry-test.ts]: [thought] This case as is in this test is not realistic. I am not sure why we're using this approach (direct DOM injection of a symbol in the root container) instead of using the methods provided by the service itself

File: showcase/tests/unit/services/hds-icon-registry-test.ts [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+      'warns once when sprite root is missing',
+    );
+    assert.strictEqual(
+      service.getSymbolId(definition),
+      null,
+      'icon remains unresolved when no sprite root exists',
+    );
+  });
+
+  test('it generates a symbol id key without size for carbon library icons', async function (assert) {
```
Comment (didoo) [inline showcase/tests/unit/services/hds-icon-registry-test.ts]: [nit]
```suggestion
  test('it generates a symbol id key without size for Carbon library icons', async function (assert) {
```

File: showcase/tests/unit/services/hds-icon-registry-test.ts [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+    const definition = makeDefinition({
+      library: HdsIconLibraryValues.Carbon,
```
Comment (didoo) [inline showcase/tests/unit/services/hds-icon-registry-test.ts:373]: [issue?] I have changed this line to `library: HdsIconLibraryValues.Flight,` and the test was passing anyway, so I am not sure what this test is doing

File: showcase/tests/unit/services/hds-icon-registry-test.ts (hunk 3 above)
Comment (didoo) [reply showcase/tests/unit/services/hds-icon-registry-test.ts]: I asked Copilot in my VSCode and that's what it told me:

> Given your confirmation that only the service populates the sprite root, these DOM checks are genuinely redundant — the in-memory _entries map is always the source of truth. If a symbol is in the DOM, the service put it there, meaning the entry already exists in _entries and requestLoad would have returned early on the this._entries.has(key) check anyway.
>
> The _symbolExists method and both of its call sites can be removed from the service, and this test can be deleted. The behavior it was testing — "don't re-load an already-present icon" — is already covered by the de-duplication test via the registry check.

So we should consider removing this test ("it uses an existing symbol in the DOM without invoking the loader") entirely.

**Review by didoo — COMMENTED** [review]
Left a few comments

File: showcase/tests/unit/services/hds-icon-registry-test.ts (hunk 3 above)
Comment (zamoore) [reply showcase/tests/unit/services/hds-icon-registry-test.ts]: That's fair - It was an edge case for simulating pre-existing sprite content outside of the service instance, but that's outside of an expected/typical use-case.

File: showcase/tests/unit/services/hds-icon-registry-test.ts (hunk 5 above)
Comment (zamoore) [reply showcase/tests/unit/services/hds-icon-registry-test.ts:373]: I rewrote it to make it valuable instead of self-referential.

File: showcase/tests/unit/services/hds-icon-registry-test.ts [hunk 6]
```diff
@@ -114,6 +101,34 @@ module('Unit | Service | hds-icon-registry', function (hooks) {
     assert.ok(service.getSymbolId.bind(service), 'getSymbolId exists');
   });
 
+  test('makeDomSafeId replaces non-DOM-safe characters', function (assert) {
+    assert.strictEqual(
+      makeDomSafeId('flight-alert-16'),
+      'flight-alert-16',
+      'keeps letters, numbers, underscores, and hyphens',
```
Comment (didoo) [inline showcase/tests/unit/services/hds-icon-registry-test.ts:108]: [nit] there is no underscore in the icon name, how can we be sure it does keep them?

**Review by didoo — APPROVED** [review]
Left only a small comment, non blocker
